ledger/accountdb.go

```diff
@@ -1491,27 +1493,34 @@ func accountsNewUpdate(qabu, qabq, qaeu, qaei, qaed *sql.Stmt, addr basics.Addre
 				pad.ExtendedAssetHolding.Insert(created, delta.new.Assets)
 			}
 
+			loaded, deletedKeys := pad.ExtendedAssetHolding.Merge()
+
 			// update DB
+			for _, key := range deletedKeys {
+				_, err = qaed.Exec(key)
+				if err != nil {
+					return updatedAccounts, err
+				}
+			}
+
 			var result sql.Result
-			for i := 0; i < len(pad.ExtendedAssetHolding.Groups); i++ {
-				if pad.ExtendedAssetHolding.Groups[i].Loaded() {
-					if pad.ExtendedAssetHolding.Groups[i].AssetGroupKey != 0 { // existing entry, update
-						_, err = qaeu.Exec(
-							pad.ExtendedAssetHolding.Groups[i].Encode(),
-							pad.ExtendedAssetHolding.Groups[i].AssetGroupKey)
-						if err != nil {
-							break
-						}
-					} else {
-						// new entry, insert
-						result, err = qaei.Exec(pad.ExtendedAssetHolding.Groups[i].Encode())
-						if err != nil {
-							break
-						}
-						pad.ExtendedAssetHolding.Groups[i].AssetGroupKey, err = result.LastInsertId()
-						if err != nil {
-							break
-						}
+			for _, i := range loaded {
+				if pad.ExtendedAssetHolding.Groups[i].AssetGroupKey != 0 { // existing entry, update
+					_, err = qaeu.Exec(
+						pad.ExtendedAssetHolding.Groups[i].Encode(),
+						pad.ExtendedAssetHolding.Groups[i].AssetGroupKey)
+					if err != nil {
+						break
+					}
+				} else {
+					// new entry, insert
+					result, err = qaei.Exec(pad.ExtendedAssetHolding.Groups[i].Encode())
+					if err != nil {
+						break
+					}
+					pad.ExtendedAssetHolding.Groups[i].AssetGroupKey, err = result.LastInsertId()
+					if err != nil {
+						break
 					}
```

Comment:
looks like you're encoding the `pad.ExtendedAssetHolding.Groups[i]` in both branches.
maybe extract it so you'll encode it only in a single place ?
i.e.
```golang
encodedAssetHoldingGroups := pad.ExtendedAssetHolding.Groups[i].Encode()
```

Reply:
I'll update it in the subsequent params PR where it looks like this (to simplify rebase):
```
	for _, i := range loadedGroups {
		ag := agl.Get(i)
		if ag.Key() != 0 { // existing entry, update
			if _, err = qaeu.Exec(ag.Encode(), ag.Key()); err != nil {
				break
			}
		} else {
			// new entry, insert
			if result, err = qaei.Exec(ag.Encode()); err != nil {
				break
			}
			var key int64
			if key, err = result.LastInsertId(); err != nil {
				break
			}
			ag.SetKey(key)
		}
	}
```